olive/passes/openvino/conversion.py

```diff
@@ -55,38 +70,45 @@ def _run_for_config(
         config: Dict[str, Any],
         output_model_path: str,
     ) -> OpenVINOModelHandler:
-        import torch
-
         try:
-            from openvino.runtime import serialize
-            from openvino.tools.mo import convert_model
+            import openvino as ov
         except ImportError:
             raise ImportError("Please install olive-ai[openvino] to use OpenVINO model") from None
 
-        # output model always has ov_model as name stem
-        model_name = "ov_model"
+        if model.framework == Framework.SNPE:
+            raise ValueError("OpenVINO conversion is not supported for SNPE model.")
 
-        model_input_param = "input_model"
         input_model = model.model_path
-        if model.framework == Framework.TENSORFLOW and Path(model.model_path).is_dir():
-            model_input_param = "saved_model_dir"
         if model.framework == Framework.PYTORCH:
             input_model = model.load_model()
 
-        if config.get("extra_config") is None:
-            config["extra_config"] = {}
+        example_input = None
+        if config.get("example_input_func"):
+            example_input = self._user_module_loader.call_object(config["example_input_func"])
```

Comment:
Could it be better if we called the `example_input_func` with the `input_model` object? Like how we do for the pytorch dummy input func. That way the input func can choose to use information from the input model like `model_attributed` if it wants. Helps when we don't want to hard code values into the user scripts to make it more flexible.

Reply:
in the ut, we could then directly use `get_pytorch_model_dummy_input` as the `example_input_func` directly.

Reply:
I don't see any use case of this. Any example for this case?

Reply:
yeah 
https://github.com/microsoft/Olive/blob/main/examples/llama2/user_script.py#L22
https://github.com/microsoft/Olive/blob/main/examples/whisper/code/user_script.py#L145

Reply:
(Not remember if our team had such discussion before) I still don't want to encourage user to use OliveModelHandler in the user script (or any Olive internal implementation), as they have to have pre-knowledge of how OliveModelHandler looks like and how it works. For the llama2 example and whisper example, I can see it is possible to avoid using PyTorchModelHandler but those codes can still work after refactoring.

Reply:
If we do want to use some metadata from OliveModel, I prefer to specially expose a attribute like "metadata" or "extra_data" to user. So they don't need to read our source code of model handler but they can know they can get some data from this attribute from Olive model.

Reply:
Sure, we can discuss about it separately and just have this pass keep its current signature. 

The `model_attributes` are mean to function similar to the "metadata" you mention. We could also do something like allow the user to provide `func_kwargs` like we do for metrics https://github.com/microsoft/Olive/blob/bc8b3312251484986972cf53ac8d4b2f4ed2ca1f/examples/notebooks/llama2/config.json#L91

Reply:
Yeah, we can especially expose some model_attributes for all our models.